lib/jekyll/filters.rb

```diff
@@ -441,6 +441,12 @@ def read_liquid_attribute(liquid_data, property)
       property.split(".").reduce(liquid_data) do |data, key|
         data.respond_to?(:[]) && data[key]
       end
+    rescue StandardError => e
```

Comment:
Is StandardError too broad? Should we keep it to TypeError or only catch when we see the error message you mention in the PR description?